peers/config.go

```diff
@@ -18,4 +19,5 @@ type Config interface {
 	GetAllowPrivateIPs() bool
 	GetTrackedSubnets() set.Set[ids.ID]
 	GetTLSCert() *tls.Certificate
+	GetUpgradeConfig() *upgrade.Config
```

Comment:
So I think it might make more sense to instead have `*upgrade.Config` as a field in the `AppRequestNetwork` object, which would make it accessible everywhere. We can take the initialize func that's currently on the config and just call it in the initialization of `AppRequestNetwork` to get the config once. That way we also don't have to remember to call `Initialize` on the config, which I'm not a huge fan of.